webdriver/tests/take_element_screenshot/user_prompts.py

```diff
@@ -15,56 +15,94 @@ def take_element_screenshot(session, element_id):
         )
     )
 
+@pytest.fixture
+def check_user_prompt_closed_without_exception(session, create_dialog):
+    def check_user_prompt_closed_without_exception(dialog_type):
+        session.url = inline("<input/>")
+        element = session.find.css("input", all=False)
+
+        create_dialog(dialog_type, text=dialog_type)
+
+        response = take_element_screenshot(session, element.id)
+        value = assert_success(response)
+
+        assert_png(value)
+
+    return check_user_prompt_closed_without_exception
+
 
 @pytest.fixture
-def check_user_prompt_not_closed_without_exception(session, create_dialog):
-    def check_user_prompt_not_closed_without_exception(dialog_type):
+def check_user_prompt_closed_with_exception(session, create_dialog):
+    def check_user_prompt_closed_with_exception(dialog_type):
         session.url = inline("<input/>")
         element = session.find.css("input", all=False)
 
         create_dialog(dialog_type, text=dialog_type)
 
+        response = take_element_screenshot(session, element.id)
+        assert_error(response, "unexpected alert open")
+
+        # Since dismiss and notify or accept and notify would have dismissed,
+        # retrying the screenshot will succeed
         response = take_element_screenshot(session, element.id)
         value = assert_success(response)
 
         assert_png(value)
 
-        assert session.alert.text == dialog_type
+    return check_user_prompt_closed_with_exception
+
+
+@pytest.fixture
+def check_user_prompt_not_closed_but_exception(session, create_dialog):
+    def check_user_prompt_not_closed_but_exception(dialog_type):
+        session.url = inline("<input/>")
+        element = session.find.css("input", all=False)
+
+        create_dialog(dialog_type, text=dialog_type)
+
+        response = take_element_screenshot(session, element.id)
+        assert_error(response, "unexpected alert open")
+
         session.alert.dismiss()
+        response = take_element_screenshot(session, element.id)
+        value = assert_success(response)
```

Comment:
Please also take the code path of other user prompt tests here, which is checking for the alert text first, and then dismissing the alert. There is no need to run a second command too.

Reply:
Fixed.